shell.nix

```diff
@@ -1,15 +1,23 @@
 { pkgs ? import (fetchGit {
     url = https://github.com/dapphub/dapptools.git;
-    ref = dapptoolsRev;
+    ref = dapptools-rev;
   }) {}
-, dapptoolsRev ? "d8e78aedaaeda323fb583ea52bef250634399e6a"
+, dapptools-rev ? "d8e78aedaaeda323fb583ea52bef250634399e6a"
```

Comment:
I used `dapptoolsRev` to distinguish from cases in which the actual name contains a `-`, e.g. `mcd-cli`

Reply:
I see, might be good practice.

Thinking of removing this as an arg and just have the rev in the fetch statement. What do you think?

Reply:
Would make it harder to override. Do you think this would never be useful?

Reply:
Overriding is definitely useful it's just that having two arguments for one dependency might be overkill and also potentially confusing, e.g. ```nix-shell --arg pkgs 'import ../dapptools {}' --argstr dapptoolsRev some-rev-hash```, this would be misleading as the `dapptoolsRev` arguments would be ignored.

You will still be able to do ```nix-shell --arg pkgs 'import (fetchGit { url = ...; rev = ...; }) {}'```, it's as flexible but a bit more consistent while sacrificing some convenience.

I'm not super convinced myself, but I think having fewer arguments and just writing `url` and `rev` in `shell.nix` also looks a bit cleaner.

Reply:
Fine by me. We'll see if this becomes annoying in practice, for the moment I've never overridden it on the CLI.